### PR TITLE
system/libuv: Bump to v1.42.0

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
Bump libuv to latest release with new porting. depends on https://github.com/apache/incubator-nuttx/pull/5612

Note: In this porting, we use standard POSIX system call instead of some nuttx's specific logic (they may have some benefits for code size or memory overhead)
## Impact
libuv
## Testing
Tested with adb on my custom board.

### CODE REVIEW

File: system/libuv/0001-unix-check-wehter-SIGPROF-exists.patch [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+diff --git a/src/unix/loop.c b/src/unix/loop.c
+index a88e71c3..dad463b3 100644
+--- a/src/unix/loop.c
++++ b/src/unix/loop.c
+@@ -220,8 +220,10 @@ int uv__loop_configure(uv_loop_t* loop, uv_loop_option option, va_list ap) {
+   if (option != UV_LOOP_BLOCK_SIGNAL)
+     return UV_ENOSYS;
+ 
++#ifdef SIGPROF
```
Comment (xiaoxiang781216) [inline system/libuv/0001-unix-check-wehter-SIGPROF-exists.patch:21]: the change is general, let's upstream to libuv community.

File: system/libuv/0003-nuttx-initial-support.patch [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+diff --git a/include/uv/unix.h b/include/uv/unix.h
+index e3cf7bdd..bf8dc32c 100644
+--- a/include/uv/unix.h
++++ b/include/uv/unix.h
+@@ -71,6 +71,8 @@
+ # include "uv/posix.h"
+ #elif defined(__QNX__)
+ # include "uv/posix.h"
++#elif defined(__NuttX__)
```
Comment (xiaoxiang781216) [inline system/libuv/0003-nuttx-initial-support.patch:27]: let's upstream NuttX specific change to libuv community too.

**Comment by masayuki2009** [comment]
@no1wudi 

```
====================================================================================
Configuration/Tool: stm32f4discovery/adb,CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Disabling CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
  Enabling CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
  Building NuttX...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   127  100   127    0     0    651      0 --:--:-- --:--:-- --:--:--   651

100  3490    0  3490    0     0   9457      0 --:--:-- --:--:-- --:--:--  9457
100 1619k    0 1619k    0     0  3056k      0 --:--:-- --:--:-- --:--:--  9.8M
From https://github.com/spiriou/microADB
 * branch            b7025c67b866925d1e64c016a844a6a3392557a4 -> FETCH_HEAD
In file included from /github/workspace/sources/apps/system/libuv/libuv/include/uv.h:66,
                 from libuv/src/unix/core.c:21:
Error: /github/workspace/sources/apps/system/libuv/libuv/include/uv/unix.h:382:18: error: field 'orig_termios' has incomplete type
  382 |   struct termios orig_termios;                                                \
      |                  ^~~~~~~~~~~~
```

File: system/libuv/Kconfig [hunk 3]
```diff
@@ -17,20 +17,29 @@ config LIBUV_THREADPOOL_SIZE
 	int "libuv default thread pool size"
 	default 1
 
-config LIBUV_TEST
-	tristate "Enable uv_run_tests"
-	default n
+choice
+	prompt "libuv utils"
```
Comment (xiaoxiang781216) [inline system/libuv/Kconfig:20]: add
default LIBUV_UTILS_NONE

File: system/libuv/Kconfig [hunk 4]
```diff
@@ -17,20 +17,29 @@ config LIBUV_THREADPOOL_SIZE
...
+
+config LIBUV_UTILS_NONE
```
Comment (xiaoxiang781216) [inline system/libuv/Kconfig:22]: merge to the first patch too

File: system/libuv/Kconfig [hunk 5]
```diff
@@ -6,7 +6,6 @@
 config LIBUV
 	bool "libuv asynchronous I/O Library"
 	default n
-	select CLOCK_MONOTONIC
```
Comment (xiaoxiang781216) [inline system/libuv/Kconfig]: merge to the first patch too

File: system/libuv/Makefile [hunk 6]
```diff
@@ -20,7 +20,7 @@
 
 include $(APPDIR)/Make.defs
 
-LIBUV_PATCHS ?= $(sort $(wildcard $(LIBUV_DIR)/000*.patch))
+LIBUV_PATCHS ?= $(sort $(wildcard 000*.patch))
```
Comment (xiaoxiang781216) [inline system/libuv/Makefile:23]: merge to the first patch too

File: system/libuv/Makefile [hunk 7]
```diff
@@ -86,6 +86,7 @@ CSRCS += pipe.c
 CSRCS += fs.c
 CSRCS += fs-poll.c
 CSRCS += timer.c
+CSRCS += process.c
```
Comment (xiaoxiang781216) [inline system/libuv/Makefile:89]: merge

**Comment by xiaoxiang781216** [comment]
This PR need the patch which remove -D__NUTTX__ from system/adb/Makefile to pass CI.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**